src/olympia/api/tests/test_authentication.py

```diff
@@ -221,3 +221,8 @@ def test_verify(self):
         assert response.status_code == 200
         data = json.loads(response.content)
         assert data['token'] == token
+
+    def test_invalid_user_id(self):
+        token = self.client.generate_api_token(self.user, user_id=-1)
+        with self.assertRaises(AuthenticationFailed):
+            self._authenticate(token)
```

Comment:
Now that you moved the `user.deleted` logic out of DRF-JWT and into AMO, I think you need a test for that.